src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -1779,16 +1831,17 @@ public void addSavedEndpoint(InetAddressAndPort ep)
         if (epState != null)
         {
             logger.debug("not replacing a previous epState for {}, but reusing it: {}", ep, epState);
-            epState.setHeartBeatState(new HeartBeatState(0));
+            epState.setHeartBeatState(HeartBeatState.empty());
         }
         else
         {
-            epState = new EndpointState(new HeartBeatState(0));
+            epState = new EndpointState(HeartBeatState.empty());
+            logger.info("Adding {} as there was no previous epState; new state is {}", ep, epState);
         }
 
         epState.markDead();
         endpointStateMap.put(ep, epState);
-        unreachableEndpoints.put(ep, System.nanoTime());
+        markDead(ep, epState);
```

Comment:
at this point in the startup no listeners exist, I was mostly thinking to avoid having the same logic multiple places.  I do see your point about logs as this could spam larger clusters; will refactor.